--- a/dlls/ucrtbase/ucrtbase.spec
+++ b/dlls/ucrtbase/ucrtbase.spec
@@ -2043,7 +2043,7 @@
 @ cdecl _wcstof_l(wstr ptr ptr)
 @ cdecl -ret64 _wcstoi64(wstr ptr long)
 @ cdecl -ret64 _wcstoi64_l(wstr ptr long ptr)
-@ stub _wcstoimax_l
+@ cdecl -ret64 _wcstoimax_l(wstr ptr long ptr) _wcstoi64_l
 @ cdecl _wcstol_l(wstr ptr long ptr)
 @ cdecl _wcstold_l(wstr ptr ptr) _wcstod_l
 @ cdecl -ret64 _wcstoll_l(wstr ptr long ptr) _wcstoi64_l
@@ -2053,7 +2053,7 @@
 @ cdecl -ret64 _wcstoui64_l(wstr ptr long ptr)
 @ cdecl _wcstoul_l(wstr ptr long ptr)
 @ cdecl -ret64 _wcstoull_l(wstr ptr long ptr) _wcstoui64_l
-@ stub _wcstoumax_l
+@ cdecl -ret64 _wcstoumax_l(wstr ptr long ptr) _wcstoui64_l
 @ cdecl _wcsupr(wstr)
 @ cdecl _wcsupr_l(wstr ptr)
 @ cdecl _wcsupr_s(wstr long)
@@ -2348,7 +2348,7 @@
 @ cdecl ilogbf(float)
 @ cdecl ilogbl(double) ilogb
 @ cdecl -ret64 imaxabs(int64)
-@ stub imaxdiv
+@ cdecl -norelay imaxdiv(int64 int64) lldiv
 @ cdecl is_wctype(long long) iswctype
 @ cdecl isalnum(long)
 @ cdecl isalpha(long)
@@ -2582,7 +2582,7 @@
 @ cdecl wcsstr(wstr wstr)
 @ cdecl wcstod(wstr ptr)
 @ cdecl wcstof(ptr ptr)
-@ stub wcstoimax
+@ cdecl -ret64 wcstoimax(wstr ptr long) _wcstoi64
 @ cdecl wcstok(wstr wstr ptr)
 @ cdecl wcstok_s(ptr wstr ptr)
 @ cdecl wcstol(wstr ptr long)
@@ -2592,7 +2592,7 @@
 @ cdecl wcstombs_s(ptr ptr long wstr long)
 @ cdecl wcstoul(wstr ptr long)
 @ cdecl -ret64 wcstoull(wstr ptr long) _wcstoui64
-@ stub wcstoumax
+@ cdecl -ret64 wcstoumax(wstr ptr long) _wcstoui64
 @ cdecl wcsxfrm(ptr wstr long)
 @ cdecl wctob(long)
 @ cdecl wctomb(ptr long)
--- a/dlls/msvcr120/msvcr120.spec
+++ b/dlls/msvcr120/msvcr120.spec
@@ -1892,7 +1892,7 @@
 @ cdecl _wcstof_l(wstr ptr ptr)
 @ cdecl -ret64 _wcstoi64(wstr ptr long)
 @ cdecl -ret64 _wcstoi64_l(wstr ptr long ptr)
-@ stub _wcstoimax_l
+@ cdecl -ret64 _wcstoimax_l(wstr ptr long ptr) _wcstoi64_l
 @ cdecl _wcstol_l(wstr ptr long ptr)
 @ cdecl _wcstold_l(wstr ptr ptr) _wcstod_l
 @ cdecl -ret64 _wcstoll_l(wstr ptr long ptr) _wcstoi64_l
@@ -1902,7 +1902,7 @@
 @ cdecl -ret64 _wcstoui64_l(wstr ptr long ptr)
 @ cdecl _wcstoul_l(wstr ptr long ptr)
 @ cdecl -ret64 _wcstoull_l(wstr ptr long ptr) _wcstoui64_l
-@ stub _wcstoumax_l
+@ cdecl -ret64 _wcstoumax_l(wstr ptr long ptr) _wcstoui64_l
 @ cdecl _wcsupr(wstr)
 @ cdecl _wcsupr_l(wstr ptr)
 @ cdecl _wcsupr_s(wstr long)
@@ -2212,7 +2212,7 @@
 @ cdecl ilogbf(float)
 @ cdecl ilogbl(double) ilogb
 @ cdecl -ret64 imaxabs(int64)
-@ stub imaxdiv
+@ cdecl -norelay imaxdiv(int64 int64) lldiv
 @ cdecl is_wctype(long long) iswctype
 @ cdecl isalnum(long)
 @ cdecl isalpha(long)
@@ -2472,7 +2472,7 @@
 @ cdecl wcsstr(wstr wstr)
 @ cdecl wcstod(wstr ptr)
 @ cdecl wcstof(ptr ptr)
-@ stub wcstoimax
+@ cdecl -ret64 wcstoimax(wstr ptr long) _wcstoi64
 @ cdecl wcstok(wstr wstr)
 @ cdecl wcstok_s(ptr wstr ptr)
 @ cdecl wcstol(wstr ptr long)
@@ -2482,7 +2482,7 @@
 @ cdecl wcstombs_s(ptr ptr long wstr long)
 @ cdecl wcstoul(wstr ptr long)
 @ cdecl -ret64 wcstoull(wstr ptr long) _wcstoui64
-@ stub wcstoumax
+@ cdecl -ret64 wcstoumax(wstr ptr long) _wcstoui64
 @ cdecl wcsxfrm(ptr wstr long)
 @ cdecl wctob(long)
 @ cdecl wctomb(ptr long)
--- a/dlls/msvcr120_app/msvcr120_app.spec
+++ b/dlls/msvcr120_app/msvcr120_app.spec
@@ -1586,7 +1586,7 @@
 @ cdecl _wcstof_l(wstr ptr ptr) msvcr120._wcstof_l
 @ cdecl -ret64 _wcstoi64(wstr ptr long) msvcr120._wcstoi64
 @ cdecl -ret64 _wcstoi64_l(wstr ptr long ptr) msvcr120._wcstoi64_l
-@ stub _wcstoimax_l
+@ cdecl -ret64 _wcstoimax_l(wstr ptr long ptr) msvcr120._wcstoimax_l
 @ cdecl _wcstol_l(wstr ptr long ptr) msvcr120._wcstol_l
 @ cdecl _wcstold_l(wstr ptr ptr) msvcr120._wcstold_l
 @ cdecl -ret64 _wcstoll_l(wstr ptr long ptr) msvcr120._wcstoll_l
@@ -1596,7 +1596,7 @@
 @ cdecl -ret64 _wcstoui64_l(wstr ptr long ptr) msvcr120._wcstoui64_l
 @ cdecl _wcstoul_l(wstr ptr long ptr) msvcr120._wcstoul_l
 @ cdecl -ret64 _wcstoull_l(wstr ptr long ptr) msvcr120._wcstoull_l
-@ stub _wcstoumax_l
+@ cdecl -ret64 _wcstoumax_l(wstr ptr long ptr) msvcr120._wcstoumax_l
 @ cdecl _wcsupr(wstr) msvcr120._wcsupr
 @ cdecl _wcsupr_l(wstr ptr) msvcr120._wcsupr_l
 @ cdecl _wcsupr_s(wstr long) msvcr120._wcsupr_s
@@ -1877,7 +1877,7 @@
 @ cdecl ilogbf(float) msvcr120.ilogbf
 @ cdecl ilogbl(double) msvcr120.ilogbl
 @ cdecl -ret64 imaxabs(int64) msvcr120.imaxabs
-@ stub imaxdiv
+@ cdecl -norelay imaxdiv(int64 int64) msvcr120.imaxdiv
 @ cdecl isalnum(long) msvcr120.isalnum
 @ cdecl isalpha(long) msvcr120.isalpha
 @ cdecl isblank(long) msvcr120.isblank
@@ -2135,7 +2135,7 @@
 @ cdecl wcsstr(wstr wstr) msvcr120.wcsstr
 @ cdecl wcstod(wstr ptr) msvcr120.wcstod
 @ cdecl wcstof(ptr ptr) msvcr120.wcstof
-@ stub wcstoimax
+@ cdecl -ret64 wcstoimax(wstr ptr long) msvcr120.wcstoimax
 @ cdecl wcstok(wstr wstr) msvcr120.wcstok
 @ cdecl wcstok_s(ptr wstr ptr) msvcr120.wcstok_s
 @ cdecl wcstol(wstr ptr long) msvcr120.wcstol
@@ -2145,7 +2145,7 @@
 @ cdecl wcstombs_s(ptr ptr long wstr long) msvcr120.wcstombs_s
 @ cdecl wcstoul(wstr ptr long) msvcr120.wcstoul
 @ cdecl -ret64 wcstoull(wstr ptr long) msvcr120.wcstoull
-@ stub wcstoumax
+@ cdecl -ret64 wcstoumax(wstr ptr long) msvcr120.wcstoumax
 @ cdecl wcsxfrm(ptr wstr long) msvcr120.wcsxfrm
 @ cdecl wctob(long) msvcr120.wctob
 @ cdecl wctomb(ptr long) msvcr120.wctomb
